internal/mining/cpuminer/cpuminer.go

```diff
@@ -815,11 +860,10 @@ out:
 		}
 	}
 
-	// Disable discrete mining mode and return the results.
 	log.Tracef("Generated %d blocks", len(blockHashes))
-	m.Lock()
-	m.discreteMining = false
-	m.Unlock()
+	if genCtx.Err() != nil {
+		return nil, ErrCancelDiscreteMining
```

Comment:
If any blocks have been generated perhaps this should also return their hashes alongside the error?

Reply:
This is true and something I thought about, but [JSON-RPC specifically disallows returning any information when there is an error](https://www.jsonrpc.org/specification_v1#a1.2Response), so I don't see any point in returning anything given it really only exists for that purpose.

> result - The Object that was returned by the invoked method. **This must be null in case there was an error invoking the method.**